app/qml/map/MMRecordingTools.qml

```diff
@@ -110,56 +110,30 @@ Item {
     lineBorderWidth: 0
   }
 
-  MMHighlight {
-    id: handlesHighlight
-
-    height: root.map.height
-    width: root.map.width
+  Loader {
+    active: __inputUtils.isLineLayer( __activeLayer.vectorLayer ) ||
+            __inputUtils.isPolygonLayer( __activeLayer.vectorLayer )
 
-    mapSettings: root.map.mapSettings
-    geometry: __inputUtils.transformGeometryToMapWithLayer( mapTool.handles, __activeLayer.vectorLayer, root.map.mapSettings )
-
-    lineStrokeStyle: ShapePath.DashLine
-    lineWidth: MMHighlight.LineWidths.Narrow
+    sourceComponent: handlesHighlightComponent
   }
 
-  MMHighlight {
-    id: guideline
+  Loader {
+    active: __inputUtils.isLineLayer( __activeLayer.vectorLayer ) ||
+            __inputUtils.isPolygonLayer( __activeLayer.vectorLayer )
 
-    height: root.map.height
-    width: root.map.width
-
-    lineWidth: MMHighlight.LineWidths.Narrow
-    lineStrokeStyle: ShapePath.DashLine
-
-    mapSettings: root.map.mapSettings
-    geometry: guidelineController.guidelineGeometry
+    sourceComponent: guidelineHighlightComponent
   }
 
-  MMHighlight {
-    id: midSegmentsHighlight
-
-    height: root.map.height
-    width: root.map.width
+  Loader {
+    active: __inputUtils.isLineLayer( __activeLayer.vectorLayer ) ||
+            __inputUtils.isPolygonLayer( __activeLayer.vectorLayer )
 
-    mapSettings: root.map.mapSettings
-    geometry: __inputUtils.transformGeometryToMapWithLayer( mapTool.midPoints, __activeLayer.vectorLayer, root.map.mapSettings )
-
-    markerType: MMHighlight.MarkerTypes.Circle
-    markerBorderColor: __style.grapeColor
+    sourceComponent: midSegmentsHighlightComponent
   }
 
-  MMHighlight {
-    id: existingVerticesHighlight
-
-    height: root.map.height
-    width: root.map.width
-
-    mapSettings: root.map.mapSettings
-    geometry: __inputUtils.transformGeometryToMapWithLayer( mapTool.existingVertices, __activeLayer.vectorLayer, root.map.mapSettings )
-
-    markerType: MMHighlight.MarkerTypes.Circle
-    markerSize: MMHighlight.MarkerSizes.Bigger
+  Loader {
+    active: true
+    sourceComponent: existingVerticesHighlightComponent
   }
```

Comment:
We do not need to have the loader altogether if we want to show it all the time :)